docs/site/exposing-graphql-apis.md

```diff
@@ -0,0 +1,131 @@
+---
+lang: en
+title: 'Exposing GraphQL APIs'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/exposing-graphql-apis.html
+---
+
+## Overview
+
+The [OASGraph module](https://www.npmjs.com/package/oasgraph) creates a GraphQL
+wrapper for existing REST APIs which are described by the OpenAPI specification.
+This tutorial shows how to expose GraphQL APIs in an existing LoopBack
+application.
+
+### Prerequisite
+
+Make sure you have a running LoopBack 4 application. In this tutorial, we'll use
+the `todo` example. You can create this application by running the command
+below:
+
+```sh
+lb4 example todo
+```
+
+### Customize the Server URL of the REST Server
+
+OASGraph expects the provided OpenAPI specification to have the `servers` url to
+be the server hosting the REST endpoints. We're going to modify the constructor
+of `TodoListApplication` in `src/application.ts` to include this information.
+
+```ts
+constructor(options: ApplicationConfig = {}) {
+    // Add this snippet at the beginning of the constructor
+    options = Object.assign(
+      {},
+      {
+        rest: {
+          openApiSpec: {
+            servers: [{url: 'http://localhost:3000'}],
+          },
+        },
+      },
+    );
+    // end of the code snippet
+    super(options);
+    ...
+}
+```
+
+### Install OASGraph and Required Dependencies
+
+From your LoopBack application, run the following command to install OASGraph
+and the required dependencies:
+
+```sh
+npm i --save oasgraph express-graphql
```

Comment:
I am curious, why do we have to install `express-graphql` ourselves? I would expect the CLI command `oasgraph` to be self-contained and come with all dependencies.

Reply:
The **OASGraph** is meant to be used as a library to transform from **OpenAPI** to **GraphQL** schema, It doesn't need `express-graphql`, since you can use any other package once you have your schema converted.  The main idea of the library  is to be used inside the application.

However the **CLI** is used to test the implementation and it requires `express-graphql`. It is not a separate package _(like monorepo)_, it is just a single file that implements the **GraphQL** server with the generated schema. IMO, it is not meant for production.

Reply:
Thanks @marioestradarosa. @dhmlau has pointed me to https://github.com/strongloop/oasgraph/issues/30, I have a better understanding now.

So, assuming oasgraph is not going to change, I am proposing the following changes:

Include `express` in the modules to install, we must not rely on `express` being installed in top-level `node_modules` by being a dependency of `@loopback/rest`.

```shell
$ npm i --save oasgraph express express-graphql
```

Explain the users why we are asking them to install those two extra express modules and point them to the discussion in https://github.com/strongloop/oasgraph/issues/30 to learn more and/or give us feedback.